assets/js/amazon-wc-checkout.js

```diff
@@ -79,6 +79,51 @@
 			renderButton( '#pay_with_amazon_cart', 'cart' );
 		}
 
+		/* Handles Amazon Pay Button on Product Pages. */
+		if ( $( '#pay_with_amazon_product' ).length > 0 ) {
+			var amazonProductBtn = renderButton( '#pay_with_amazon_product', 'product' );
+			if ( null !== amazonProductBtn ) {
+				amazonProductBtn.onClick( function() {
+					if ( $( '.single_add_to_cart_button' ).hasClass( 'disabled' ) ) {
+						$( '.single_add_to_cart_button' ).trigger( 'click' );
+						return;
+					}
+					var elemToBlock = $( '#pay_with_amazon_product' ).closest( 'div.summary' ).length > 0 ? $( '#pay_with_amazon_product' ).closest( 'div.summary' ) : false;
+					if ( elemToBlock ) {
+						block( elemToBlock );
+					}
+					var pid = $( 'button[type="submit"][name="add-to-cart"]' ).attr( 'value' );
+					pid = pid || $( 'input[type="hidden"][name="product_id"]' ).val();
+					var qnt = $( 'input[type="number"][name="quantity"]' ).val();
+					var vid = $( 'input[type="hidden"][name="variation_id"]' ).length > 0 ? $( 'input[type="hidden"][name="variation_id"]' ).val() : false;
+					var qrs = '&pid=' + pid + '&qnt=' + qnt + ( vid ? '&vid=' + vid : '' );
```

Comment:
Here is a snippet from another project where we need to gather data related to what is being added to cart. 
```
		const $thisbutton = $( this ),
			$form = $thisbutton.closest( 'form.cart' ),
			data = {
				action: 'my_action',
			};

		if ( $( this ).is( '.disabled' ) ) {
			return false;
		}

		const id = $thisbutton.val();

		$.each( $form.serializeArray(), function( index, object ) {
			if ( 'add-to-cart' === object.name ) {
				data.product_id = object.value;
			} else {
				data[ object.name ] = object.value;
			}
		} );

		data.quantity = data.quantity || 1;
		data.variation_id = data.variation_id || id;
		data.product_id = data.product_id || id;
```

Not a blocker but it is slightly less reliant on the exact markup.

Reply:
Used a similar approach as the one provided.